sdk/azidentity/chained_token_credential.go

```diff
@@ -14,12 +14,20 @@ import (
 
 // ChainedTokenCredentialOptions contains optional parameters for ChainedTokenCredential.
 type ChainedTokenCredentialOptions struct {
-	// placeholder for future options
+	// RetrySources configures how the credential uses its sources.
+	// When true, the credential will always request a token from each source in turn,
+	// stopping when one provides a token. When false, the credential requests a token
+	// only from the source that previously retrieved a token--it never again tries the sources which failed.
+	RetrySources bool
```

Comment:
I doubt we'll find a perfect name for this, but I wonder whether we can come up with a more self-explanatory one. It controls whether `GetToken` always iterates over the sources, which has me thinking along the lines of `AlwaysIterate`, `AlwaysStartWithFirstSource`, or `AlwaysStartAtBeginning`. It would be nice if this option paired with the (opposite) option in other SDKs, so something like DoNot/PreferFirstSuccessfulSource or Forget/RememberFirstSuccessfulSource, but that's probably too high a bar. I'm just trying to start a brainstorm here.

Reply:
What about `cacheSuccessfulCredential` ?

Reply:
It would be good to name it in such a way that its zero-value (false) is the preferred behavior.  Calling it `CacheSuccessfulCredential` would mean we'd need to set it to `true` by default.  I think `RetrySources` is reasonable, however we could rename it before release.